report/3.Theory/SyntaxTheory.tex

```diff
@@ -0,0 +1,178 @@
+		R                 & The set of all real numbers             \\ \hline
+		D                 & \{$Z^+ \cup Z^-$\}                      \\ \hline
+		$S_p$             & The set of all special characters      \\ \hline
+		$L_l$             & The set of all lower-case letters       \\ \hline
+		$L_u$             & The set of all upper-case letters       \\ \hline
+		C                 & $S_p \cup L_l \cup L_u \cup D$          \\ \hline
+		$R_{eserved}$     & The set of all reserved keywords in HCL \\ \hline
+	\end{tabular}
+	\caption{Table containing all utility alphabets used to formally describe the type and identifier languages}
+\end{table}
+
+By using the set construction method, the languages can now be defined as shown in Table 3.2.
+
+\begin{table}[!htb]
+	\centering
+	\label{my-label}
+	\begin{tabular}{|l|l|}
+		\hline
+		\textbf{Name}     & \textbf{Set}                                    \\ \hline
+		$L_{none}$        & \{w | w $\in$ \{none\}\}                        \\ \hline
+		$L_{bool}$        & \{w | w $\in$ (T $\cup$ F)\}                    \\ \hline
+		$L_{number}$      & \{w | w $\in$ R\}                               \\ \hline
+		$L_{text}$        & \{"w" | all strings from arbitrary alphabets\}  \\ \hline
+		$L_{identifiers}$ & \{w | w $\in (L_{text} - \{", "\}) \wedge w \notin R_{eserved}$\} \\ \hline
+	\end{tabular}
+	\caption{Table containing the type and identifier alphabets described by the set builder construction}
+\end{table}
+
+Intuitively it is trivial that all of the languages defined above are regular languages.
+This is the case since a regular language cannot keep track of more than one symbol in terms of counting and iterations.
+Since all of the languages in Table 3.2 consists only of arbitrary concatenations of elements of various alphabets, this means that they are indeed regular. 
+Now that it has been determined that the languages are regular, a computational model can now be chosen to describe them more specifically.
+A regular expression for each of the languages will now be constructed.
+
+For the sake of convenience $R^+$ will be used at a shorthand for $RR^*$ \footnote{Where * is Kleene star}, meaning that $R^+ \cup \epsilon = R^*$ since $R^*$ consists of all strings that are zero or more concatenations from R and $RR^*$ consists of all strings that are one or more concatenations of strings from R.
+The resulting regular expressions are shown in Table 3.3.
+
+\begin{table}[!htb]
+	\centering
+	\label{my-label}
+	\begin{tabular}{|l|l|}
+		\hline
+		\textbf{Name}     & \textbf{Regular Expression}         \\ \hline
+		$L_{none}$        & none                                \\ \hline
+		$L_{bool}$        & true $\cup$ false                   \\ \hline
+		$L_{number}$      & $D^+(.D^+)^*$                       \\ \hline
+		$L_{text}$        & $"C^*"$                               \\ \hline
+		$L_{identifiers}$ & $C^+(R_{eserved} \cup \epsilon)C^+$ \\ \hline
```

Comment:
R_eserved?
wut?

Reply:
it is consistent so we will stick with it